### PR TITLE
Add ignore_if_bech32_mismatch to restore_backup()

### LINKED ISSUE #87: Correctly generate address and transaction history from backup
## Description

When importing a mainnet stronghold backup into a testnet profile. `recover_accounts` returns `smr...` on first start-up, additionally, the transaction history is ported from mainnet to testnet.

## Motivation

Don't bleed mainnet state over into testnet state whenever recovered.

## Requirements

Write a list of what you want this feature to do.

1. If there has been a transaction history on mainnet with active funds
2. And it is being imported in a testnet profile
3. Reset the stronghold state (addresses, outputs,...), so no mainnet state is stored in testnet profiles

## Steps to reproduce

1) Create a mainnet profile
2) Do some transactions on mainnet
3) Create a stronghold backup
4) Create a testnet profile recovered from a stronghold backup

## Are you planning to do it yourself in a pull request?
No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add ignore_if_bech32_mismatch to restore_backup()

## Links to any relevant issues

Fixes #87 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Added test, with a modified nodejs example

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/core/src/method/wallet.rs [hunk 1]
```diff
@@ -118,6 +118,7 @@ pub enum WalletMethod {
         #[derivative(Debug(format_with = "OmittedDebug::omitted_fmt"))]
         password: String,
         ignore_if_coin_type_mismatch: Option<bool>,
+        ignore_if_bech32_mismatch: Option<String>,
```
Comment (thibault-martinez) [inline bindings/core/src/method/wallet.rs:129]: Also add the doc you added to the TS file?

File: bindings/nodejs/lib/wallet/Wallet.ts [hunk 2]
```diff
@@ -252,13 +252,15 @@ export class Wallet {
         source: string,
         password: string,
         ignoreIfCoinTypeMismatch?: boolean,
+        ignoreIfBech32Mismatch?: string,
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/wallet/Wallet.ts:257]: Also add the doc here?

File: sdk/src/wallet/message_interface/message.rs [hunk 3]
```diff
@@ -111,6 +111,7 @@ pub enum Message {
         /// Stronghold file password.
         password: String,
         ignore_if_coin_type_mismatch: Option<bool>,
+        ignore_if_bech32_mismatch: Option<String>,
```
Comment (thibault-martinez) [inline sdk/src/wallet/message_interface/message.rs:122]: Add the doc?

File: sdk/tests/wallet/backup_restore.rs [hunk 4]
```diff
@@ -433,3 +439,85 @@ async fn backup_and_restore_different_coin_type_dont_ignore() -> Result<()> {
 
     tear_down(storage_path)
 }
+
+#[tokio::test]
+#[cfg(all(feature = "stronghold", feature = "storage"))]
+// Backup and restore with Stronghold
```
Comment (thibault-martinez) [inline sdk/tests/wallet/backup_restore.rs]: Comment before attributes

File: sdk/src/wallet/message_interface/message_handler.rs [hunk 5]
```diff
@@ -389,9 +395,15 @@ impl WalletMessageHandler {
         backup_path: PathBuf,
         stronghold_password: String,
         ignore_if_coin_type_mismatch: Option<bool>,
+        ignore_if_bech32_mismatch: Option<&str>,
```
Comment (thibault-martinez) [inline sdk/src/wallet/message_interface/message_handler.rs]: Why do we need backup/restore_backup as function here? They just call the real one it feels useless?

Comment (Thoralf-M) [reply sdk/src/wallet/message_interface/message_handler.rs]: It is useless, removed that already in the new bindings

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**